### PR TITLE
pkg: archive: don't fail Untar if xattrs are not supported [rhel7-1.10.3]

### PR DESCRIPTION
**Pull request by runcom**
@rhatdan @maxamillion @mrunalp PTAL 

This is probably meant as just a workaround - upstream issue is tracking this whole thing here https://github.com/docker/docker/issues/20331

Fix https://bugzilla.redhat.com/show_bug.cgi?id=1343702

Since certain filesystems don't support extended attributes, ignore
errors produced (emitting a warning) when attempting to apply extended
attributes to file.

Signed-off-by: Aleksa Sarai <asarai@suse.de>

### CODE REVIEW

File: pkg/archive/archive.go [hunk 1]
```diff
@@ -425,10 +425,19 @@ func createTarFile(path, extractDir string, hdr *tar.Header, reader io.Reader, L
...
 	}
 
+	var errors []string
 	for key, value := range hdr.Xattrs {
 		if err := system.Lsetxattr(path, key, []byte(value), 0); err != nil {
-			return err
+			// We ignore errors here because not all graphdrivers support xattrs.
+			errors = append(errors, err.Error())
 		}
+
```
Comment (rhatdan) [inline pkg/archive/archive.go]: Shouldn't they by checking err == ENOTSUPP?

Comment (runcom) [inline pkg/archive/archive.go]: Not sure, this patch has already been accepted upstream, do you think upstream did wrong?

Comment (rhatdan) [inline pkg/archive/archive.go]: I just think they should only ignore if it is ENOTSUPP, if it was EPERM or some other error like outofspace, probably should still fail.

**Comment by rhatdan** [comment]
Merging anyways we should open discussion upstream.

File: pkg/archive/archive.go (hunk 1 above)
Comment (rhatdan) [inline pkg/archive/archive.go]: Do you have a link to where this patch is happening upstream?

Comment (runcom) [inline pkg/archive/archive.go]: it already happened here https://github.com/docker/docker/pull/21251

**Pull request merged**